src/gasPriceOracle/oracle.ts

```diff
@@ -96,18 +142,22 @@ export async function getViemGasPriceEstimate(
   let maxFeePerGas: bigint;
   let maxPriorityFeePerGas: bigint;
   if (gasPriceFeeds[chainId]) {
-    ({ maxFeePerGas, maxPriorityFeePerGas } = await gasPriceFeeds[chainId](viemProvider, chainId));
+    ({ maxFeePerGas, maxPriorityFeePerGas } = await gasPriceFeeds[chainId](
+      viemProvider,
+      chainId,
+      baseFeeMultiplier,
+      unsignedTx
+    ));
   } else {
     let gasPrice: bigint | undefined;
     ({ maxFeePerGas, maxPriorityFeePerGas, gasPrice } = await viemProvider.estimateFeesPerGas());
 
-    maxFeePerGas ??= gasPrice!;
+    maxFeePerGas ??= gasPrice! * BigInt(baseFeeMultiplier);
     maxPriorityFeePerGas ??= BigInt(0);
   }
 
```

Comment:
This is fwiw where I think we could unconditionally apply the baseFeeMultiplier (as mentioned above):

`maxFeePerGas = (maxFeePerGas - maxPriorityFeePerGas) * baseMultiplier + maxPriorityFeePerGas;`

I think applying it here should simplify and declutter all of the adapters, since they won't need to be concerned with the baseFeeMultiplier at all.

Reply:
But doesn't that mean that we're assuming all chains add priority fee to base fee to get full base fee? Is that always right?

Reply:
Correct; and that's not valid on Linea :headstone:. That's pretty annoying...

Reply:
Sorry I don't understand why is that not valid on linea?